[CI] Reproducible Failure in org.elasticsearch.indices.recovery.RecoveryTests.testTranslogHistoryTransferred"\

Just happened to me in a PR but also reproduces locally:

```
REPRODUCE WITH: ./gradlew 'null' --tests "org.elasticsearch.indices.recovery.RecoveryTests.testTranslogHistoryTransferred" -Dtests.seed=776EA849CF005E89 -Dtests.locale=uk-UA -Dtests.timezone=Pacific/Guadalcanal -Druntime.java=14
```

fails as:

```
[2021-02-01T23:49:56,485][INFO ][o.e.i.r.RecoveryTests    ] [testTranslogHistoryTransferred] before test
[2021-02-01T23:50:21,692][WARN ][o.e.i.r.RecoverySourceHandler] [org.elasticsearch.indices.recovery.RecoveryTests] [test][0][recover to s1] 0 Remote file corruption on node {s1}{s1}{6iQc54zQTBS3KC0orCJkJQ}{0.0.0.0}{0.0.0.0:3}{dimr}, recovering [name [_0.cfe], length [405], checksum [19dwde5], writtenBy [8.8.0]]. local checksum OK
org.apache.lucene.index.CorruptIndexException: checksum failed (hardware problem?) : expected=19dwde5 actual=1032yzu (resource=name [_0.cfe], length [405], checksum [19dwde5], writtenBy [8.8.0]) (resource=VerifyingIndexOutput(_0.cfe))
	at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.readAndCompareChecksum(Store.java:1196) ~[classes/:?]
	at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeByte(Store.java:1174) ~[classes/:?]
	at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeBytes(Store.java:1204) ~[classes/:?]
	at org.elasticsearch.indices.recovery.MultiFileWriter.innerWriteFileChunk(MultiFileWriter.java:133) ~[classes/:?]
	at org.elasticsearch.indices.recovery.MultiFileWriter$FileChunkWriter.writeChunk(MultiFileWriter.java:227) ~[classes/:?]
	at org.elasticsearch.indices.recovery.MultiFileWriter.writeFileChunk(MultiFileWriter.java:78) ~[classes/:?]
	at org.elasticsearch.indices.recovery.RecoveryTarget.writeFileChunk(RecoveryTarget.java:502) [classes/:?]
	at org.elasticsearch.indices.recovery.AsyncRecoveryTarget.lambda$writeFileChunk$6(AsyncRecoveryTarget.java:88) [classes/:?]
	at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:680) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
[2021-02-01T23:50:21,736][INFO ][o.e.i.r.RecoveryTests    ] [testTranslogHistoryTransferred] after test
REPRODUCE WITH: ./gradlew 'null' --tests "org.elasticsearch.indices.recovery.RecoveryTests.testTranslogHistoryTransferred" -Dtests.seed=776EA849CF005E89 -Dtests.locale=uk-UA -Dtests.timezone=Pacific/Guadalcanal -Druntime.java=14

java.lang.AssertionError: org.elasticsearch.indices.recovery.RecoveryFailedException: [test][0]: Recovery failed from {s0}{s0}{wvu2YTYzQRiciU64EA3vvg}{0.0.0.0}{0.0.0.0:2}{dimr} into {s1}{s1}{6iQc54zQTBS3KC0orCJkJQ}{0.0.0.0}{0.0.0.0:3}{dimr}

	at __randomizedtesting.SeedInfo.seed([776EA849CF005E89:41178B987FB29090]:0)
	at org.elasticsearch.index.shard.IndexShardTestCase$2.onRecoveryFailure(IndexShardTestCase.java:133)
	at org.elasticsearch.indices.recovery.RecoveryTarget.notifyListener(RecoveryTarget.java:266)
	at org.elasticsearch.indices.recovery.RecoveryTarget.fail(RecoveryTarget.java:253)
	at org.elasticsearch.index.shard.IndexShardTestCase.recoverUnstartedReplica(IndexShardTestCase.java:654)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase.access$700(ESIndexLevelReplicationTestCase.java:112)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationGroup.recoverReplica(ESIndexLevelReplicationTestCase.java:420)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationGroup.recoverReplica(ESIndexLevelReplicationTestCase.java:411)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationGroup.recoverReplica(ESIndexLevelReplicationTestCase.java:406)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationGroup.startReplicas(ESIndexLevelReplicationTestCase.java:271)
	at org.elasticsearch.index.replication.ESIndexLevelReplicationTestCase$ReplicationGroup.startAll(ESIndexLevelReplicationTestCase.java:261)
	at org.elasticsearch.indices.recovery.RecoveryTests.testTranslogHistoryTransferred(RecoveryTests.java:83)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:824)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:475)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
	at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
	at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
	at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:375)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:831)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: org.elasticsearch.indices.recovery.RecoveryFailedException: [test][0]: Recovery failed from {s0}{s0}{wvu2YTYzQRiciU64EA3vvg}{0.0.0.0}{0.0.0.0:2}{dimr} into {s1}{s1}{6iQc54zQTBS3KC0orCJkJQ}{0.0.0.0}{0.0.0.0:3}{dimr}
	... 45 more
Caused by: org.elasticsearch.transport.RemoteTransportException: [File corruption occurred on recovery but checksums are ok]
	Suppressed: org.apache.lucene.index.CorruptIndexException: checksum failed (hardware problem?) : expected=19dwde5 actual=1032yzu (resource=name [_0.cfe], length [405], checksum [19dwde5], writtenBy [8.8.0]) (resource=VerifyingIndexOutput(_0.cfe))
		at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.readAndCompareChecksum(Store.java:1196)
		at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeByte(Store.java:1174)
		at org.elasticsearch.index.store.Store$LuceneVerifyingIndexOutput.writeBytes(Store.java:1204)
		at org.elasticsearch.indices.recovery.MultiFileWriter.innerWriteFileChunk(MultiFileWriter.java:133)
		at org.elasticsearch.indices.recovery.MultiFileWriter$FileChunkWriter.writeChunk(MultiFileWriter.java:227)
		at org.elasticsearch.indices.recovery.MultiFileWriter.writeFileChunk(MultiFileWriter.java:78)
		at org.elasticsearch.indices.recovery.RecoveryTarget.writeFileChunk(RecoveryTarget.java:502)
		at org.elasticsearch.indices.recovery.AsyncRecoveryTarget.lambda$writeFileChunk$6(AsyncRecoveryTarget.java:88)
		at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingRunnable.run(ThreadContext.java:680)
		at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
		at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
		at java.base/java.lang.Thread.run(Thread.java:832)

```

https://gradle-enterprise.elastic.co/s/whnbotirrvizq/console-log/raw

Looking into it.

--- a/server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java
+++ b/server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java
@@ -941,9 +941,10 @@ public class RecoverySourceHandler {
                 @Override
                 protected void executeChunkRequest(FileChunk request, ActionListener<Void> listener) {
                     cancellableThreads.checkForCancel();
+                    final ReleasableBytesReference content = new ReleasableBytesReference(request.content, request);
                     recoveryTarget.writeFileChunk(
-                        request.md, request.position, ReleasableBytesReference.wrap(request.content), request.lastChunk,
-                            translogOps.getAsInt(), ActionListener.runBefore(listener, request::close));
+                        request.md, request.position, content, request.lastChunk,
+                            translogOps.getAsInt(), ActionListener.runBefore(listener, content::close));
                 }
 
                 @Override

